catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/security/WorkspacePolicyExtension.java

```diff
@@ -36,105 +33,102 @@
-        return Stream.of(new KeyValuePermission(systemUserAttribute, ImmutableSet.of(
-                systemUserAttributeValue)));
+    private Map<String, Set<String>> groupPermissionsByKey(List<KeyValuePermission> permissions) {
+        return permissions.stream()
+                .collect(Collectors.toMap(KeyValuePermission::getKey,
+                        KeyValuePermission::getValues,
+                        Sets::union));
     }
 
-    @Override
-    public KeyValueCollectionPermission isPermittedMatchAll(CollectionPermission subject,
-            KeyValueCollectionPermission match) {
-
-        List<KeyValuePermission> permissions = getPermissions(match);
-
-        Stream<KeyValuePermission> remapped = permissions.stream()
-                .filter(byKeys(keyMapping.keySet()))
-                .map(remapKeys(keyMapping));
-
-        boolean matched = concat(overrides(), remapped).filter(subject::implies)
-                .findFirst()
-                .isPresent();
-
-        if (!matched) {
-            return match;
-        }
-
-        if (isOwnerPermission(getPermissions(match), subject)) {
-            return new KeyValueCollectionPermission();
-        }
-
-        return new KeyValueCollectionPermission(match.getAction(),
-                permissions.stream()
-                        .filter(byKeys(keyMapping.keySet()).negate())
-                        .collect(Collectors.toList()));
+    private Predicate<CollectionPermission> systemPred() {
+        Set<String> values = ImmutableSet.of(config.getSystemUserAttributeValue());
+        KeyValuePermission perm = new KeyValuePermission(config.getSystemUserAttribute(), values);
+        return (subject) -> subject.implies(perm);
     }
 
-    private boolean isOwnerPermission(List<KeyValuePermission> permissions,
-            CollectionPermission subject) {
-        Optional<KeyValuePermission> ownerPerm = permissions.stream()
-                .filter(keyValuePermission -> keyValuePermission.getKey()
-                        .equals(Core.METACARD_OWNER))
-                .findAny();
-
-        if (ownerPerm.isPresent()) {
-            KeyValuePermission ownerEmailPermission =
-                    new KeyValuePermission(Constants.EMAIL_ADDRESS_CLAIM_URI,
-                            ownerPerm.get()
-                                    .getValues());
-            return subject.implies(ownerEmailPermission);
+    private Predicate<CollectionPermission> pred(Map<String, Set<String>> p, String k1, String k2) {
+        if (!p.containsKey(k1)) {
+            return (subject) -> false;
         }
-        return false;
+        return (subject) -> subject.implies(new KeyValuePermission(k2, p.get(k1)));
     }
 
-    @Override
-    public KeyValueCollectionPermission isPermittedMatchOne(CollectionPermission subject,
-            KeyValueCollectionPermission matchOne) {
-        return matchOne;
+    private Predicate<CollectionPermission> ownerPred(Map<String, Set<String>> permissions) {
```

Comment:
✏️ Your choice, but you don't really need `Pred` on the end since the return type should indicate what is getting returned. (as for the other ones as well if you want)

Reply:
I do like shorter names, and it is in the method signature. Removed.